pkg/manager/member/tidb_member_manager.go

```diff
@@ -393,12 +393,19 @@ func getTiDBConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
 	}
 
 	plugins := tc.Spec.TiDB.Plugins
-	startScript, err := RenderTiDBStartScript(&TidbStartScriptModel{
-		ClusterName:     tc.Name,
+	tidbStartScriptModel := &TidbStartScriptModel{
 		EnablePlugin:    len(plugins) > 0,
 		PluginDirectory: "/plugins",
 		PluginList:      strings.Join(plugins, ","),
-	})
+	}
+
+	if tc.IsHeterogeneous() {
+		tidbStartScriptModel.Path = controller.PDMemberName(tc.Spec.Cluster.Name) + ":2379"
+	} else {
+		tidbStartScriptModel.Path = controller.PDMemberName(tc.Name) + ":2379"
```

Comment:
This will trigger the rolling update of TiDB as it was `${CLUSTER_NAME}-pd:2379` before.
Please help confirm, if yes, we have to fix this in another PR including master.

Reply:
This problem fix in pr #3160

Reply:
#3160 only fixed the issue for TiKV.